From f936515a3b4d76b1da25102cc1079d29d670ae65 Mon Sep 17 00:00:00 2001
From: Alexandru Costache <alexandru@balena.io>
Date: Wed, 14 Dec 2022 12:20:12 +0100
Subject: [PATCH] imx8m_var_dart:  Use custom_fdt_file if defined

his assumes the Supervisor adds the custom device tree name
to extra_uEnv.txt. The extra_uEnv.txt contents would be
custom_fdt_file=custom_device_tree.dtb.

If unset, the board continues to load the default device-tree.

Upstream-status: Inaproppriate [configuration]
Signed-off-by: Alexandru Costache <alexandru@balena.io>
---
 include/configs/imx8mq_var_dart.h | 24 ++++++++++++++----------
 1 file changed, 14 insertions(+), 10 deletions(-)

diff --git a/include/configs/imx8mq_var_dart.h b/include/configs/imx8mq_var_dart.h
index f02ac2ce09..36f0b11cf7 100644
--- a/include/configs/imx8mq_var_dart.h
+++ b/include/configs/imx8mq_var_dart.h
@@ -132,18 +132,22 @@
 		"fi;\0" \
 	"findfdt=" \
 		"if test $fdt_file = undefined; then " \
-			"if test $carrier_rev = legacy; then " \
-				"if test ${mmcdev} = 0; then " \
-					"setenv fdt_file imx8mq-var-dart-dt8mcustomboard-legacy-wifi-hdmi.dtb; " \
-				"else " \
-					"setenv fdt_file imx8mq-var-dart-dt8mcustomboard-legacy-sd-hdmi.dtb; " \
-				"fi; " \
+			"if test -n ${custom_fdt_file}; then " \
+				"setenv fdt_file ${custom_fdt_file}; " \
 			"else " \
-				"if test ${mmcdev} = 0; then " \
-					"setenv fdt_file imx8mq-var-dart-dt8mcustomboard-wifi-hdmi.dtb; " \
+				"if test $carrier_rev = legacy; then " \
+					"if test ${mmcdev} = 0; then " \
+						"setenv fdt_file imx8mq-var-dart-dt8mcustomboard-legacy-wifi-hdmi.dtb; " \
+					"else " \
+						"setenv fdt_file imx8mq-var-dart-dt8mcustomboard-legacy-sd-hdmi.dtb; " \
+					"fi; " \
 				"else " \
-					"setenv fdt_file imx8mq-var-dart-dt8mcustomboard-sd-hdmi.dtb; " \
-                                "fi; " \
+					"if test ${mmcdev} = 0; then " \
+						"setenv fdt_file imx8mq-var-dart-dt8mcustomboard-wifi-hdmi.dtb; " \
+					"else " \
+						"setenv fdt_file imx8mq-var-dart-dt8mcustomboard-sd-hdmi.dtb; " \
+					"fi; " \
+				"fi; " \
 			"fi; " \
 		"fi; \0" \
 	"loadfdt=run findfdt; " \
-- 
2.37.2

